src/screens/DiscoveryScreenV2/elements/Search.js

```diff
@@ -289,17 +280,30 @@ const styles = StyleSheet.create({
     marginRight: 11,
     ...FONTS.h3
   },
-  animatedViewContainer: {
+  animatedViewContainer: (hideBackIcon) => ({
     flexDirection: 'row',
     backgroundColor: 'white',
     marginTop: 0,
     zIndex: 10,
     height: dimen.size.DISCOVERY_HEADER_HEIGHT,
     paddingTop: 7,
     paddingBottom: 7,
-    borderBottomWidth: 1,
+    borderBottomWidth: hideBackIcon ? 0 : 1,
     borderBottomColor: COLORS.alto
-  }
+  })
 });
 
+DiscoverySearch.propTypes = {
+  setDiscoveryLoadingData: PropTypes.func,
+  searchText: PropTypes.string,
+  setSearchText: PropTypes.func,
+  placeholderText: PropTypes.string,
+  setIsFocus: PropTypes.func,
+  setIsFirstTimeOpen: PropTypes.func,
+  fetchDiscoveryData: PropTypes.func,
+  fetchData: PropTypes.func,
+  onCancelToken: PropTypes.func,
+  hideBackIcon: PropTypes.bool
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

PropTypes have been added for all props used in this component. This is a good practice as it helps catch bugs during development by validating the types of props.
```diff
+ DiscoverySearch.propTypes = {
+   setDiscoveryLoadingData: PropTypes.func,
+   searchText: PropTypes.string,
+   setSearchText: PropTypes.func,
+   placeholderText: PropTypes.string,
+   setIsFocus: PropTypes.func,
+   setIsFirstTimeOpen: PropTypes.func,
+   fetchDiscoveryData: PropTypes.func,
+   fetchData: PropTypes.func,
+   onCancelToken: PropTypes.func,
+   hideBackIcon: PropTypes.bool
+ };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->